airbyte-cdk/bulk/core/load/src/test/kotlin/io/airbyte/cdk/state/CheckpointManagerTest.kt

```diff
@@ -345,9 +346,14 @@ class CheckpointManagerTest {
 
     @ParameterizedTest
     @ArgumentsSource(CheckpointManagerTestArgumentsProvider::class)
-    fun testAddingAndFlushingCheckpoints(testCase: TestCase) {
+    suspend fun testAddingAndFlushingCheckpoints(testCase: TestCase) = runTest {
         if (testCase.expectedException != null) {
-            Assertions.assertThrows(testCase.expectedException) { runTestCase(testCase) }
+            try {
```

Comment:
`assertThrows`

Reply:
You can't pass a suspend function to it.